src/hotspot/share/runtime/deoptimization.cpp

```diff
@@ -728,12 +728,11 @@ JRT_LEAF(BasicType, Deoptimization::unpack_frames(JavaThread* thread, int exec_m
   // must point to the vframeArray for the unpack frame.
   vframeArray* array = thread->vframe_array_head();
 
-#ifndef PRODUCT
   if (TraceDeoptimization) {
     tty->print_cr("DEOPT UNPACKING thread " INTPTR_FORMAT " vframeArray " INTPTR_FORMAT " mode %d",
                   p2i(thread), p2i(array), exec_mode);
+    tty->cr();
```

Comment:
Same question as above about the necessity of `cr()`.

Reply:
```
DEOPT PACKING thread 0x00007f9b9b008a20 Compiled frame (sp=0x0000700002121f00 unextended sp=0x0000700002121f00, fp=0x0000000000000000, real_fp=0x0000700002121f30, pc=0x000000011bad5944)
     nmethod   1714  154       4       java.lang.String::indexOf (29 bytes)
     Virtual frames (innermost first):
        0 - (0x00007f9b9ff91038) - ifge @ bci 13 
        1 - (0x00007f9b9ff923a0) - invokestatic @ bci 13 
     Created vframeArray 0x00007f9ba280a820

DEOPT UNPACKING thread 0x00007f9b9b008a20 vframeArray 0x00007f9ba280a820 mode 2

{method} {0x000000012d00b8c8} 'indexOf' '(II)I' in 'java/lang/String' - invokestatic @ bci 13 sp = 0x0000700002121eb8
Expressions size: 0
Locals size: 3
[1. Interpreted Frame]
[...]
```
I have put it for separating the `DEOPT PACKING`, `DEOPT UNPACKING` and `SAVED OOP RESULT` blocks for readability reasons. But I don't mind removing it, if a compact log is preferred.

Reply:
I don't feel strongly about it but I think `-XX:+TraceDeoptimization -XX:-PrintDeoptimizationDetails` has never printed blank lines. However, this is primarily for human consumption so it's fine to leave as you have it.